js/hitbtc3.js

```diff
@@ -2083,6 +2083,41 @@ module.exports = class hitbtc3 extends Exchange {
         return this.safeNumber (response, 'leverage');
     }
 
+    async setLeverage (leverage, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a symbol argument');
+        }
+        if (params['margin_balance'] === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a margin_balance parameter that will transfer margin to the specified trading pair');
+        }
+        const market = this.market (symbol);
+        const amount = this.safeNumber (params, 'margin_balance');
+        if (market['type'] !== 'swap') {
+            throw new BadSymbol (this.id + ' setLeverage() supports swap contracts only');
+        }
+        if (symbol === 'BTC/USDT:USDT') {
```

Comment:
Can we avoid hardcoding these symbols? Let's add them to user-controllable options:

```JavaScript
describe() {
    'options': {
        'setLeverage': {
            'maxLeverage': {
                'BTC/USDT:USDT': 100,
                'ETH/USDT:USDT': 75,
                'ADA/USDT:USDT': 75,
                'SOL/USDT:USDT': 75,
                'XRP/USDT:USDT': 75,
            },
        },
    },
}

async setLeverage (...) {
     const options = this.safeValue (this.options, 'setLeverage', {});
     const maxLeverages = this.safeValue (options, 'maxLeverage', {});
     const maxLeverage = this.safeNumber (maxLeverages, symbol, 50);
     if ((leverage < 1) || (leverage > maxLeverage)) {
         throw new BadRequest (this.id + ' setLeverage() leverage should be between 1 and ' + maxLeverage.toString () + ' for ' + symbol);
     }
     // ...
}
```